int32[] x
int32[] y
int32[] z
uint32[] cost
int32[3] map_size
int32[3] map_offset
